src/07-embedding/03-artworks.ts

```diff
@@ -0,0 +1,278 @@
+import { ARTWORK_IDS } from "./artwork-ids"
+import weaviate from "weaviate-ts-client"
+import _ from "lodash"
+
+const client = weaviate.client({
+  scheme: "https",
+  host: "https://weaviate.stg.artsy.systems",
+})
+
+const CLASS_NAME: string = "SmallNewTrendingArtworks"
+
+const BATCH_SIZE = 100
+
+async function main() {
+  const artworks = await fetchArtworks()
+
+  console.log(`Fetched ${artworks.length} artworks`)
+  console.log(artworks)
+  await prepareArtworksCollection()
+  await insertArtworks(artworks)
+  /*
+   * now query from Weaviate console:
+   * https://console.weaviate.io
+   */
+}
+
+main()
+
+async function prepareArtworksCollection() {
+  const client = weaviate.client({
+    scheme: "https",
+    host: "https://weaviate.stg.artsy.systems",
+  })
+
+  const alreadyExists = await client.schema.exists(CLASS_NAME)
+
+  if (alreadyExists) {
+    console.log(`${CLASS_NAME} class already exists, deleting it`)
+    await client.schema.classDeleter().withClassName(CLASS_NAME).do()
+  }
+
+  const classWithProps = {
+    class: CLASS_NAME,
+    vectorizer: "text2vec-openai",
+    moduleConfig: {
+      "text2vec-openai": {
+        model: "text-embedding-3-small",
```

Comment:
See slack conversation about the trade offs between using `weaviate` to vectorize for us vs [doing it ourselves](https://github.com/artsy/quantum/blob/main/src/07-embedding/01-minimal.ts#L20-L34). In short, we are much more vulnerable to model versions when depending of `weaviate`.